optuna/study/_frozen.py

```diff
@@ -0,0 +1,94 @@
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from optuna import logging
+from optuna.study._study_direction import StudyDirection
+
+
+_logger = logging.get_logger(__name__)
+
+
+class FrozenStudy:
+    """Basic attributes of a :class:`~optuna.study.Study`.
+
+    This class is private and not referenced by Optuna users.
+
+    Attributes:
+        study_name:
+            Name of the :class:`~optuna.study.Study`.
+        direction:
+            :class:`~optuna.study.StudyDirection` of the :class:`~optuna.study.Study`.
+
+            .. note::
+                This attribute is only available during single-objective optimization.
+        directions:
+            A list of :class:`~optuna.study.StudyDirection` objects.
+        user_attrs:
+            Dictionary that contains the attributes of the :class:`~optuna.study.Study` set with
+            :func:`optuna.study.Study.set_user_attr`.
+        system_attrs:
+            Dictionary that contains the attributes of the :class:`~optuna.study.Study` internally
+            set by Optuna.
+
+    """
+
+    def __init__(
+        self,
+        study_name: str,
+        direction: Optional[StudyDirection],
+        user_attrs: Dict[str, Any],
+        system_attrs: Dict[str, Any],
+        study_id: int,
+        *,
+        directions: Optional[List[StudyDirection]] = None,
```

Comment:
Sorry for the nit-picky comment, but we prefer `Sequence` when we specify the type of arguments. (c.f., https://github.com/optuna/optuna/blob/f0c4f380d9b7c3a86a1b15b952ee6e09b1451904/optuna/study/study.py#L1033)

I guess the code actually accept a sequence of `StudyDirection` since we apply `list` function to `directions` at L51.

```suggestion
        directions: Optional[Sequence[StudyDirection]] = None,
```